src/components/tag/Tag.root.tsx

```diff
@@ -0,0 +1,63 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { Spacing } from '../../foundations/layout';
+import { applyPadding } from '../../foundations/layout/Grid.mixins';
+import { applyShape } from '../../foundations/shape/Shape.mixins';
+import { TTheme } from '../../foundations/theme-provider/themes/theme.types';
+import { applyTextMargin, applyTextStyles, applyTextColor } from '../text/Text.mixins';
+import Text from '../text';
+import { Utils } from '../../shared';
+
+import { PTagRoot } from './Tag.props';
+import { TTagVariant } from './Tag.types';
+
+const TagRoot = styled(Text).withConfig({
+    shouldForwardProp: Utils.forwardProperties(),
+})<PTagRoot>(
+    ({ variant, size, theme }: ThemedStyledProps<PTagRoot, TTheme>): FlattenSimpleInterpolation => {
+        const isMention = variant === 'highlight';
+
+        const TAG_BACKGROUND_COLOR_MAP: Record<TTagVariant, TTheme> = {
+            general: theme.background.skeleton,
+            info: theme.palette.primary.light,
+            warning: theme.palette.alert.light,
+            success: theme.palette.success.light,
+            highlight: theme.highlight.mention,
+            shortcut: theme.background.skeleton,
+        };
+
+        const colors: Record<string, string> = {
+            background: TAG_BACKGROUND_COLOR_MAP[variant],
+            text: isMention ? theme.palette.primary.main : theme.text.primary,
+        };
+
+        return css`
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            background-color: ${colors.background};
+            color: ${colors.text};
+            text-transform: ${isMention ? 'none' : 'uppercase'};
```

Comment:
since components should be generic IMO this should be handled from the outside (like pass in a already "uppercased" string)

Reply:
I think in this case, since the label itself handles the uppercase it should be handled in this component.